controllers/Frontend.php

```diff
@@ -1,61 +1,133 @@
+
+                
+                $errors = [];
+
+                if(empty($name) || empty($email) || empty($subject) || empty($message)){
+                    $errors['empty'] = "Veuillez remplier tous les champs";
+                }
+                if(strlen($name) < 5){
+                    $errors['name'] = "Votre nom doit contenir au moins 5 caractères.";
+                }
+                if(strlen($subject) < 5){
+                    $errors['subject'] = "Votre sujet doit contenir au moins 5 caractères.";
+                }
+                if(strlen($message) < 5){
+                    $errors['message'] = "Votre message doit contenir au moins 50 caractères.";
+                }
+
+                if(empty($errors)){
+                    contact_mail($name,$email,$subject,$message);
+                    contact_mail_user($name,$email,$subject,$message);
+                }
+            }
+            else{
+                $errors = [];
+            }
         
-        $page="frontend/home_cv";
-        $topbar="frontend/topbar";
-        require "views/layout.php";
+        $myView = new View('home_cv', 'frontend');
+        $myView->render(array('errors' => $errors));
     }
 
     public function legalnotice()
     {
-        $page="frontend/legalnotice";
-        $topbar="frontend/topbar";
-        require "views/layout.php";
+        $myView = new View('legalnotice', 'frontend');
+        $myView->render();
     }
     
     public function home()
     {
-        $model_article = new Article();
-        $model_comment = new Comment();
-
-        $page="frontend/home";
-        $topbar="frontend/topbar";
-        require "views/layout.php";
+        $manager_article = new ArticleManager();
+        $posts = $manager_article->get_posts_blog();
+        
+        $manager_user = new UserManager();
+        $admins = $manager_user->get_posts_blog_admin();
+        
+        $myView = new View('home', 'frontend');
+        $myView->render(array('posts' => $posts, 'admins' => $admins));
     }
 
     public function blog()
     {
-        $model_article = new Article();
-        $model_comment = new Comment();
+        $manager_article = new ArticleManager();
+        $posts = $manager_article->get_posts_blog_all();
         
-        $page="frontend/blog";
-        $topbar="frontend/topbar";
-        require "views/layout.php";
+        $manager_user = new UserManager();
+        $admins = $manager_user->get_posts_blog_all_admin();
+        
+        $myView = new View('blog', 'frontend');
+        $myView->render(array('posts' => $posts, 'admins' => $admins));
     }
 
     public function post()
     {
-        $model_article = new Article();
-        $model_comment = new Comment();
-                
-        $page="frontend/post";
-        $topbar="frontend/topbar";
-        require "views/layout.php";
+        if(filter_var(filter_input(INPUT_GET,'id')))
+        {
+            $id = filter_input(INPUT_GET, 'id');
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $id. Configured minimum length is 3.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)